sampler.go

```diff
@@ -355,18 +390,19 @@ func (s *adaptiveSampler) update(strategies *sampling.PerOperationSamplingStrate
 	for _, strategy := range strategies.PerOperationStrategies {
 		operation := strategy.Operation
 		samplingRate := strategy.ProbabilisticSampling.SamplingRate
-		lowerBound := strategies.DefaultLowerBoundTracesPerSecond
+		minSamplesPerSecond := strategies.DefaultLowerBoundTracesPerSecond
 		if sampler, ok := s.samplers[operation]; ok {
-			sampler.update(lowerBound, samplingRate)
+			sampler.update(minSamplesPerSecond, samplingRate, defaultMaxSamplesPerSecond)
 		} else {
-			sampler := newGuaranteedThroughputProbabilisticSampler(
-				lowerBound,
+			s.samplers[operation] = newGuaranteedThroughputProbabilisticSampler(
+				minSamplesPerSecond,
 				samplingRate,
+				defaultMaxSamplesPerSecond,
 			)
-			s.samplers[operation] = sampler
 		}
 	}
-	s.lowerBound = strategies.DefaultLowerBoundTracesPerSecond
+	s.minSamplesPerSecond = strategies.DefaultLowerBoundTracesPerSecond
+	s.maxSamplesPerSecond = defaultMaxSamplesPerSecond // TODO (wjang) get this from strategies
```

Comment:
do you want to land first a change upgrading the strategy type to include maxSamplesPerSecond?